vendor/processors/stripe.py

```diff
@@ -1,104 +1,261 @@
 """
 Payment processor for Stripe.
 """
-from django.conf import settings
-
+import logging
 import stripe
-
+from django.conf import settings
 from .base import PaymentProcessorBase
-from vendor.models.choice import PurchaseStatus
+from vendor.models.choice import (
+    SubscriptionStatus,
+    TransactionTypes,
+    PaymentTypes,
+    TermType,
+    TermDetailUnits,
+    InvoiceStatus,
+    PurchaseStatus
+)
+from vendor.integrations import StripeIntegration
+
+logger = logging.getLogger(__name__)
 
 class StripeProcessor(PaymentProcessorBase):
+    """
+    Implementation of Stripe SDK
+    https://stripe.com/docs/api/authentication?lang=python
+    """
 
-    def processor_setup(self):
-        stripe.api_key = settings.STRIPE_TEST_PUBLIC_KEY    # TODO: This should work, but may not the best way to do this
+    TRANSACTION_SUCCESS_MESSAGE = 'message'
+    TRANSACTION_SUCCESS_CODE = 'code'
+    TRANSACTION_FAIL_MESSAGE = 'error_text'
+    TRANSACTION_FAIL_CODE = 'error_code'
+    TRANSACTION_RESPONSE_CODE = 'response_code'
+    transaction_submitted = False
 
-    def get_checkout_context(self, request=request, context={}):
-        '''
-        The Invoice plus any additional values to include in the payment record.
-        '''
-        context = super().get_checkout_context(request=request, context=context)
-        context['integration_check'] = 'accept_a_payment'
+    products_mapping = {}
 
-        metadata = {}
-        metadata['integration_check'] = 'accept_a_payment'
-        metadata['order_id'] = str(self.invoice.pk)
+    def get_checkout_context(self, request=None, context={}):
+        context = super().get_checkout_context(context=context)
+        # TODO need to figure out how we're building stripe form
+        """if 'credit_card_form' not in context:
+            context['credit_card_form'] = CreditCardForm(initial={'payment_type': PaymentTypes.CREDIT_CARD})
+        if 'billing_address_form' not in context:
+            context['billing_address_form'] = BillingAddressForm()"""
+        return context
```

Comment:
You can remove this function as we already know that our form has the necessary information to create payments and subscriptions